### PR TITLE
[SWE-1537][BRANDBOOK MODULE] Write Switch component

### PR DESCRIPTION
**Pull request by ArtemSuhoy**

### CODE REVIEW

**Comment by ArtemSuhoy** [comment]
image

File: src/app/components/elements/Switcher/Switch.styled.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    position: absolute;
+    top: 0;
+    left: 0;
+    right: 0;
+    bottom: 0;
+    background-color: var(--brandbook-module-switch-bg-default);
+    -webkit-transition: var(--brandbook-module-switch-transition);
+    transition: var(--brandbook-module-switch-transition);
+    border-radius: inherit;
+    cursor: ${({ checked }) => (checked ? 'default' : 'pointer')};
```
Comment (aleksandrgolyk) [inline src/app/components/elements/Switcher/Switch.styled.js:75]: Why you change cursor on checked? Swither always must be pointed

File: src/app/components/elements/Switcher/Switch.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  return (
+    <SwitchStyled variant={variant} checked={checked}>
+      <input type="checkbox" checked={checked} />
+      <span className="slider" />
+    </SwitchStyled>
+  )
+}
+
+Switch.propTypes = {
+  variant: PropTypes.string,
```
Comment (alexpassss) [inline src/app/components/elements/Switcher/Switch.js:16]: PropTypes.oneOf(['primary', 'secondary'])

File: src/app/components/elements/Switcher/Switch.styled.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
+import styled, { css } from 'styled-components'
+
+const SWITCH_VARIANT = {
+  default: css`
```
Comment (alexpassss) [inline src/app/components/elements/Switcher/Switch.styled.js:4]: same situation with objects

File: src/app/components/elements/Switcher/Switch.styled.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
+import styled, { css } from 'styled-components'
+
+const SWITCH_VARIANT = {
```
Comment (alexpassss) [inline src/app/components/elements/Switcher/Switch.styled.js:3]: this is size

File: src/styles/vars/switch.css [hunk 5]
```diff
@@ -0,0 +1,48 @@
+:root {
+  /* Switch */
+
+  /* Switch color */
+  --brandbook-module-switch-bg-default: #25303752;
```
Comment (alexpassss) [inline src/styles/vars/switch.css:5]: don't use hex

File: src/styles/vars/switch.css [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  --brandbook-module-switch-bg-active: var(--brandbook-module-accent);
```
Comment (alexpassss) [inline src/styles/vars/switch.css:6]: use mediator instead of direct color

File: src/styles/vars/switch.css [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  --brandbook-module-switch-border-radius-default: calc(
+    var(--brandbook-module-switch-height-default) / 2
+  );
+  --brandbook-module-switch-border-radius-mega: calc(
+    var(--brandbook-module-switch-height-mega) / 2
+  );
+
+  /* Switch slider */
+  --brandbook-module-switch-slider-bg: var(--brandbook-module-bg-white);
+  --brandbook-module-switch-slider-transition: 0.4s;
```
Comment (alexpassss) [inline src/styles/vars/switch.css:30]: group with the same line above

File: src/styles/vars/switch.css [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+  --brandbook-module-switch-slider-width-default: 14px;
+
+  --brandbook-module-switch-slider-height-mega: 280px;
+  --brandbook-module-switch-slider-width-mega: 280px;
+
+  /* Switch slider border radius */
+  --brandbook-module-switch-slider-border-radius: 50%;
+
+  /* Switch slider offsets */
+  --brandbook-module-switch-slider-left-default: 1px;
```
Comment (alexpassss) [inline src/styles/vars/switch.css:43]: same issue with var naming

File: src/app/components/elements/Switcher/Switch.js [hunk 9]
```diff
@@ -0,0 +1,20 @@
+import { SwitchStyled } from './Switch.styled'
+import PropTypes from 'prop-types'
+import React from 'react'
```
Comment (Olexandr25) [inline src/app/components/elements/Switcher/Switch.js:3]: do you need here React?

File: src/app/components/elements/Switcher/Switch.js [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+
+const Switch = (props) => {
```
Comment (Olexandr25) [inline src/app/components/elements/Switcher/Switch.js:5]: write comment - js doc

**Review by AnastasiaShapowalova — APPROVED** [review]

**Review by Olexandr25 — APPROVED** [review]

**Pull request merged**